gsc/tests/24-proc/apply.scm

```diff
@@ -0,0 +1,17 @@
+(declare (extended-bindings) (not constant-fold) (not safe))
+
+(println (##apply eq? '(1 1)))
+(println (##apply eq? '(1 2)))
+
+(println (##apply + '(1)))
+(println (##apply * '(1 2)))
+(println (##apply - '(1 2 3)))
+(println (##apply + '(1 2 3 4)))
+(println (##apply * '(1 2 3 4 5)))
+(println (##apply - '(1 2 3 4 5 6)))
+(println (##apply + '(1 2 3 4 5 6 7)))
+(println (##apply * '(1 2 3 4 5 6 7 8)))
+(println (##apply - '(1 2 3 4 5 6 7 8 9)))
+(println (##apply + '(1 2 3 4 5 6 7 8 9 10)))
+
+(println (##apply + '(1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20)))
```

Comment:
These tests aren't valid because ```eq?```, ```+```, etc are not predefined procedures at this level of testing.  So instead of ```eq?``` there should be ```(lambda (x y) (##eqv? x y))```. Of course, ```##apply``` is currently unimplemented in the universal backend but it will have to work at some point (note that lib/_univlib.scm defines a hackish ```##apply``` that works for a small number of arguments).